diff --git a/appimagebuilder/modules/deploy/apt/package.py b/appimagebuilder/modules/deploy/apt/package.py
index 792a724d..6b59d3a3 100644
--- a/usr/local/lib/python3.11/dist-packages/appimagebuilder/modules/deploy/apt/package.py
+++ b/usr/local/lib/python3.11/dist-packages/appimagebuilder/modules/deploy/apt/package.py
@@ -13,7 +13,7 @@
 import urllib
 from pathlib import Path
 
-from packaging import version
+from pydpkg import Dpkg
 
 
 class Package:
@@ -76,7 +76,7 @@ def __str__(self):
 
     def __gt__(self, other):
         if isinstance(other, Package):
-            return version.parse(self.version) > version.parse(other.version)
+            return Dpkg.compare_versions(self.version, other.version) > 0
 
     def __hash__(self):
         return self.__str__().__hash__()
