diff --git a/Makefile b/Makefile
index 3edc8c93..54cf8c37 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ UNAME_M := $(shell uname -m)
 rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
 CC_IS_CLANG := $(shell $(CC) --version | grep -q clang && echo true || echo false)
 
-PACKAGES := capstone freetype2 glfw3 libavcodec libavformat libavutil libswresample libuv zlib libcurl
+PACKAGES := capstone freetype2 glfw3 libavcodec libavformat libavutil libswresample libuv zlib libcurl fmt md4c libluv liburiparser libllhttp
 
 LOCALES := el es_ES fr ja pt_BR uk zh_CN
 
@@ -23,8 +23,6 @@ CPPFLAGS += `pkg-config --cflags $(PACKAGES)`
 CPPFLAGS += -I.
 CPPFLAGS += -Isrc
 CPPFLAGS += -Ithird_party
-CPPFLAGS += -Ithird_party/ELFIO
-CPPFLAGS += -Ithird_party/fmt/include/
 CPPFLAGS += -Ithird_party/gl3w
 CPPFLAGS += -Ithird_party/googletest/googletest/include
 CPPFLAGS += -Ithird_party/imgui
@@ -32,16 +30,10 @@ CPPFLAGS += -Ithird_party/imgui/backends
 CPPFLAGS += -Ithird_party/imgui/examples
 CPPFLAGS += -Ithird_party/imgui/misc/cpp
 CPPFLAGS += -Ithird_party/libelfin
-CPPFLAGS += -Ithird_party/llhttp
 CPPFLAGS += -Ithird_party/luajit/src
-CPPFLAGS += -Ithird_party/luv/src
-CPPFLAGS += -Ithird_party/luv/deps/lua-compat-5.3/c-api
 CPPFLAGS += -Ithird_party/magic_enum/include/magic_enum
-CPPFLAGS += -Ithird_party/md4c/src
-CPPFLAGS += -Ithird_party/multipart-parser-c
 CPPFLAGS += -Ithird_party/tracy/public
-CPPFLAGS += -Ithird_party/ucl -Ithird_party/ucl/include
-CPPFLAGS += -Ithird_party/uriparser/include
 CPPFLAGS += -Ithird_party/zep/extensions
 CPPFLAGS += -Ithird_party/zep/include
 CPPFLAGS += -Ithird_party/xbyak/xbyak
@@ -86,6 +78,8 @@ else
 endif
 
 LDFLAGS += third_party/luajit/src/libluajit.a
+LDFLAGS += -lucl
+LDFLAGS += -lmultipart
 LDFLAGS += -ldl
 LDFLAGS += -g
 
@@ -109,7 +103,6 @@ endif
 LD := $(CXX)
 
 SRCS := $(call rwildcard,src/,*.cc)
-SRCS += third_party/fmt/src/os.cc third_party/fmt/src/format.cc
 IMGUI_SRCS += $(wildcard third_party/imgui/*.cpp)
 VIXL_SRCS := $(call rwildcard, third_party/vixl/src,*.cc)
 SRCS += $(IMGUI_SRCS)
@@ -129,16 +122,10 @@ SRCS += third_party/imgui/misc/freetype/imgui_freetype.cpp
 SRCS += third_party/imgui_lua_bindings/imgui_lua_bindings.cpp
 SRCS += third_party/imgui_md/imgui_md.cpp
 SRCS += third_party/imgui_memory_editor/imgui_memory_editor.cpp
-SRCS += $(wildcard third_party/llhttp/*.c)
 SRCS += $(wildcard third_party/lpeg/*.c)
 SRCS += third_party/lua-protobuf/pb.c
 SRCS += third_party/luafilesystem/src/lfs.c
-SRCS += third_party/luv/src/luv.c
-SRCS += third_party/md4c/src/md4c.c
-SRCS += third_party/multipart-parser-c/multipart_parser.c
 SRCS += third_party/nanovg/src/nanovg.c
-SRCS += third_party/tracy/public/TracyClient.cpp
-SRCS += third_party/ucl/src/n2e_99.c third_party/ucl/src/alloc.c
-SRCS += $(wildcard third_party/uriparser/src/*.c)
 SRCS += third_party/zep/extensions/repl/mode_repl.cpp
 SRCS += $(wildcard third_party/zep/src/*.cpp)
@@ -167,8 +154,6 @@ ifeq ($(CROSS),arm64)
 endif
 SUPPORT_SRCS := src/support/file.cc src/support/mem4g.cc src/support/zfile.cc
 SUPPORT_SRCS += src/supportpsx/adpcm.cc src/supportpsx/binloader.cc src/supportpsx/ps1-packer.cc
-SUPPORT_SRCS += third_party/fmt/src/os.cc third_party/fmt/src/format.cc
-SUPPORT_SRCS += third_party/ucl/src/n2e_99.c third_party/ucl/src/alloc.c
 SUPPORT_SRCS += $(wildcard third_party/iec-60908b/*.c)
 OBJECTS := third_party/luajit/src/libluajit.a
 
diff --git a/src/main/main.cc b/src/main/main.cc
index 850617ee..cecc64f4 100644
--- a/src/main/main.cc
+++ b/src/main/main.cc
@@ -364,11 +364,12 @@ runner.init({
             luacovEnabled = false;
         }
     }
+/*
     emulator->setLua();
     s_ui->setLua(*emulator->m_lua);
     emulator->m_spu->setLua(*emulator->m_lua);
     assert(emulator->m_lua->gettop() == 0);
-
+*/
     // Starting up the whole emulator; we delay setting the GPU only now because why not.
     auto &emuSettings = emulator->settings;
     emulator->m_spu->open();
